fxa/oauth.py

```diff
@@ -239,11 +240,21 @@ def verify_token(self, token, scope=None):
             resp = None
 
         if resp is None:
-            url = '/verify'
-            body = {
-                'token': token
+            ## TODO we want to fetch https://oauth.accounts.firefox.com/.well-known/openid-configuration and then get the jwks_uri key to get the following url, but we'll just hardcode it like this for now until we are closer to production. We need to do this because oauth.accounts.firefox.com does not yet respond to .well-known, but accounts.firefox.com does; for consistency, oauth.accounts.firefox.com should respond to .well-known. 
+            
+            secret = jwt.algorithms.RSAAlgorithm.from_jwk(
```

Comment:
Hrm, it occurs to me that you're going to have to grapple with the possibility of having multiple keys here. The value you fetch from the oauth server looks like this:

```
{
  "keys": [
    { ... a key ... },
    { ... another key ... },
    ...
  ]
}
```

There will *almost always* be only one key in that list, but there might be more than one if the FxA server is currently doing a key rotation. The simplest way to deal with this is probably to just try each key in turn, something like:

```
keys = self.apiclient.get('/v1/jwks')['keys']
for key in keys:
    try:
        pubkey = jwt.algorithms.RSAAlgorithm.from_jwk(key)
        decoded = jwt.decode(token, pubkey, ...)
        ...etc...
    except jwt.exceptions.InvalidSignatureError:
        continue
else:
    raise WhateverErrorClassWeHaveForInvalidTokensIveForgottenTheName
```

Reply:
Hm, yes, this had occurred to me but I hadn't thought through the implications. Trying each in turn sounds good, thanks.